naxsi_src/naxsi_runtime.c

```diff
@@ -1133,9 +1133,11 @@ naxsi_create_log_array(ngx_http_request_ctx_t* ctx,
   sz_left -= sub;
   offset += sub;
 
-  sub = NAXSI_REQUEST_ID_SIZE << 1;
+  const u_char* req_id = naxsi_request_id(r);
+
+  sub = strlen((const char*)req_id);
```

Comment:
this is another null deref when req_id is null. just add a simple check like `req_id ? strlen((const char*)req_id) : 0`